org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/projectWizard/TemplateProjectWizardFragment.xtend

```diff
@@ -0,0 +1,519 @@
+			override getProjectTemplates() {
+				#[new HelloWorldProject]
+			}
+		}
+		
+		@ProjectTemplate(label="Hello World", icon="project_template.png", description="<p><b>Hello World</b></p>
+		<p>This is a parameterized hello world for �grammar.name�. You can set a parameter to modify the content in the generated file
+		and a parameter to set the package the file is created in.</p>")
+		final class HelloWorldProject {
+			var advanced = check("Advanced", false, "Check to enabled advanced configuration")
+			var name = combo("Name", #["Xtext", "World", "Foo", "Bar"], "The name to say 'Hello' to")
+			var path = text("Package", "�language.fileExtensions.get(0)�", "The package path to place the files in")
+		
+			override protected updateVariables() {
+				name.enabled = advanced.value
+				path.enabled = advanced.value
+				if (!advanced.value) {
+					name.value = "Xtext"
+					path.value = "�language.fileExtensions.get(0)�"
+				}
+			}
+		
+			override protected validate() {
+				if (path.value.matches('[a-z][a-z0-9_]*(/[a-z][a-z0-9_]*)*'))
+					null
+				else
+					new Status(ERROR, "Wizard", "'" + path + "' is not a valid package name")
+			}
+		
+			override generateFiles(IProjectFileGenerator generator) {
+				generator.generate(�quotes�src/�openVar�path�closeVar�/Model.�language.fileExtensions.get(0)��quotes�,
+					�quotes�
+						/*
+						 * This is an example model
+						 */
+						Hello �openVar�name�closeVar�!
+					�quotes�)
+			}
+		}
+		'''
+		file.writeTo(projectConfig.eclipsePlugin.src)
+	}
+	
+	def generateProjectCreator() {
+		val genClass = getProjectCreatorClassName.typeRef
+		val projectInfoClass = projectInfoClassName.typeRef
+		val baseClass = {
+			if (pluginProject)
+				"org.eclipse.xtext.ui.wizard.AbstractPluginProjectCreator"
+			else
+				"org.eclipse.xtext.ui.wizard.AbstractProjectCreator"
+		}.typeRef
+
+		val file = fileAccessFactory.createGeneratedJavaFile(genClass)
+		
+		file.content = '''
+		import java.util.HashMap;
+		import java.util.List;
+		import java.util.Set;
+
+		import org.eclipse.core.resources.IProject;
+		import org.eclipse.core.resources.IResource;
+		import org.eclipse.core.runtime.CoreException;
+		import org.eclipse.core.runtime.IProgressMonitor;
+		import org.eclipse.xtext.builder.EclipseResourceFileSystemAccess2;
+		import org.eclipse.xtext.generator.IFileSystemAccess;
+		import org.eclipse.xtext.generator.IFileSystemAccess2;
+		import org.eclipse.xtext.generator.IOutputConfigurationProvider;
+		import org.eclipse.xtext.generator.OutputConfiguration;
+		�IF pluginProject�
+		import org.eclipse.xtext.ui.util.PluginProjectFactory;
+		�ELSE�
+		import org.eclipse.xtext.ui.util.ProjectFactory;
+		�ENDIF�
+		import org.eclipse.xtext.ui.wizard.template.AbstractProjectTemplate;
+		import org.eclipse.xtext.ui.wizard.template.ProjectFileGenerator;
+		import com.google.common.collect.ImmutableList;
+		�IF pluginProject�
+		import com.google.common.collect.Lists;
```

Comment:
Please indent the constants parts of the templates in control structures to improve readability.